### PR TITLE
Feature: File Query Parser

### LINKED ISSUE #154: Add some kind of parser that tranforms from the query string to a FileQuery object

### PR DESCRIPTION
**Pull request by iTitus**
Closes #154

### CODE REVIEW

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+    /**
+     * Add an owner.
+     *
+     * @param owner one Owner
+     * @return this
+     */
+    public FileQueryBuilder owner(@NonNull String owner) {
+        if (owner.isEmpty()) {
+            throw new IllegalArgumentException("owner must not be empty!");
```
Comment (Segelzwerg) [inline src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: Can you add a logging line here?

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+    /**
+     * Add a file name.
+     *
+     * @param name new file name to search for
+     * @return this
+     */
+    public FileQueryBuilder name(@NonNull String name) {
+        if (name.isEmpty()) {
+            throw new IllegalArgumentException("name must not be empty!");
```
Comment (Segelzwerg) [inline src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: and here

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+    /**
+     * Add a file type.
+     *
+     * @param type new type to search for
+     * @return this
+     */
+    public FileQueryBuilder type(@NonNull String type) {
+        if (type.isEmpty()) {
+            throw new IllegalArgumentException("type must not be empty!");
```
Comment (Segelzwerg) [inline src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: and here

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+    /**
+     * Add a tag.
+     *
+     * @param tag what the file should be tagged with
+     * @return this
+     */
+    public FileQueryBuilder tag(@NonNull String tag) {
+        if (tag.isEmpty()) {
+            throw new IllegalArgumentException("tag must not be empty!");
```
Comment (Segelzwerg) [inline src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: and here

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java (hunk 1 above)
Comment (iTitus) [reply src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: no, you are doing this in your pr.

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java (hunk 2 above)
Comment (iTitus) [reply src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: no, you are doing this in your pr.

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java (hunk 3 above)
Comment (iTitus) [reply src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: no, you are doing this in your pr.

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java (hunk 4 above)
Comment (iTitus) [reply src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: no, you are doing this in your pr.

File: src/main/java/mops/businesslogic/query/FileQueryBuilder.java (hunk 1 above)
Comment (realemx) [reply src/main/java/mops/businesslogic/query/FileQueryBuilder.java]: There shouldn't be logging necessary, it's internal logic

**Pull request merged**